src/algorithms/dqns/rem_dqn.jl

```diff
@@ -0,0 +1,155 @@
+- `target_approximator`::[`AbstractApproximator`](@ref): similar to `approximator`, but used to estimate the target (the next state).
+- `loss_func`: the loss function.
+- `γ::Float32=0.99f0`: discount rate.
+- `batch_size::Int=32`
+- `update_horizon::Int=1`: length of update ('n' in n-step update).
+- `min_replay_history::Int=32`: number of transitions that should be experienced before updating the `approximator`.
+- `update_freq::Int=4`: the frequency of updating the `approximator`.
+- `ensemble_num::Int=1`: the number of ensemble approximators.
+- `target_update_freq::Int=100`: the frequency of syncing `target_approximator`.
+- `stack_size::Union{Int, Nothing}=4`: use the recent `stack_size` frames to form a stacked state.
+- `traces = SARTS`, set to `SLARTSL` if you are to apply to an environment of `FULL_ACTION_SET`.
+- `rng = Random.GLOBAL_RNG`
+"""
+function REMDQNLearner(;
+    approximator::Tq,
+    target_approximator::Tt,
+    loss_func::Tf,
+    stack_size::Union{Int,Nothing} = nothing,
+    γ::Float32 = 0.99f0,
+    batch_size::Int = 32,
+    update_horizon::Int = 1,
+    min_replay_history::Int = 32,
+    update_freq::Int = 1,
+    ensemble_num::Int = 1,
+    target_update_freq::Int = 100,
+    traces = SARTS,
+    update_step = 0,
+    rng = Random.GLOBAL_RNG,
+) where {Tq,Tt,Tf}
+    copyto!(approximator, target_approximator)
+    sampler = NStepBatchSampler{traces}(;
+        γ = γ,
+        n = update_horizon,
+        stack_size = stack_size,
+        batch_size = batch_size,
+    )
+    # Build a convex polygon to make a combination of multiple Q-value estimates as a Q-value estimate.
+    convex_polygon = rand(Float32, (1, ensemble_num))
+    convex_polygon ./= sum(convex_polygon)
+    REMDQNLearner(
+        approximator,
+        target_approximator,
+        loss_func,
+        min_replay_history,
+        update_freq,
+        update_step,
+        target_update_freq,
+        sampler,
+        ensemble_num,
+        convex_polygon,
+        rng,
+        0.0f0,
+    )
+end
+
+Flux.functor(x::REMDQNLearner) = (Q = x.approximator, Qₜ = x.target_approximator),
+y -> begin
+    x = @set x.approximator = y.Q
+    x = @set x.target_approximator = y.Qₜ
+    x
+end
+
+function (learner::REMDQNLearner)(env)
+    # In the multi-head method, we need to combine ensemble_num states and input them into the approximator. TODO: better implementation.
+    s = send_to_device(device(learner.approximator), state(env))
+    s = Flux.unsqueeze(s, ndims(s) + 1)
+    s = multivcat(s, learner.ensemble_num)
+    multi_q = learner.approximator(s)
+    q = learner.convex_polygon .* reshape(multi_q, :, learner.ensemble_num)
+    vec(sum(q, dims = 2)) |> send_to_host
+end
+
+function multivcat(arr::Array{T, N}, num::Int) where{T, N}
+    marr = Array{T, N}(undef, size(arr))
+    copyto!(marr, arr)
+    for i in 2:num
+        marr = vcat(marr, arr)
+    end
+    marr
+end
```

Comment:
It limits the input to `Array`, which makes this algorithm only works on CPU.

Also this seems inefficient to me. Copying the same state each time will be very costly. You may take a look at the implementation of [`OneHotMatrix`](https://github.com/FluxML/Flux.jl/blob/master/src/onehot.jl) and create a similar structure like `TiledArray` to avoid copying data each time. I'm not sure if there's already an implementation somewhere. I would suggest you to ask it on discourse or somewhere first before implementing it.

However, this might need some extra work and we can improve it later.